File: 2021/05/25/main.go
package main

import "strconv"

func evalRPN(tokens []string) int {
	var l []int
	for _, t := range tokens {
		if t == "+" || t == "-" || t == "*" || t == "/" {
			d := len(l) - 1
			b, a := l[d], l[d-1]
			l = l[:d-1]
			l = append(l, calc(a, b, t))
		} else {
			v, _ := strconv.Atoi(t)
			l = append(l, v)
		}
	}
	return l[0]
}

func calc(a, b int, o string) int {
	switch o {
	case "+":
		return a + b
	case "-":
		return a - b
	case "*":
		return a * b
	default:
		return a / b
	}
}

File: 2021/05/25/main.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        l = []
        for t in tokens:
            if t in "+-*/":
                a, b = l[-2],  l[-1]
                l = l[:-2]
                if t == '+':
                    l.append(a+b)
                if t == '-':
                    l.append(a-b)
                if t == '*':
                    l.append(a*b)
                if t == '/':
                    l.append(int(a/b))
            else:
                l.append(int(t))
        return l[0]
